src/librbd/AioObjectRequest.cc

```diff
@@ -623,6 +624,13 @@ void AioObjectTruncate::send_write() {
   }
 }
 
+void AioObjectTrim::pre_object_map_update(uint8_t *new_state) {
+  *new_state = OBJECT_PENDING;
+
+  // object map state (before pre) decides if a post updation is needed
+  need_post = m_ictx->object_map->update_required(m_object_no, *new_state);
```

Comment:
I think that's kind of implicit. `->update_required()` in `send_pre()` would become a no-op for other cases. `AioObjectTrim::pre_object_map_update()` does something similar and sets `need_post` accordingly. Actually, I had initially thought to introduce `AioObejctTrim::send_post` that would set `need_post` thereby avoiding redundant call to `->update_required` and avoiding calling `pre_object_map_update()` call inside `object_map_lock`.